### ISSUE TITLE
WebWolf no email

### ISSUE THREAD

**Issue by brohan** [WebGoat/WebGoat#454]
Using docker (ubuntu18.04) I Issued to following commands:
WebGoat:
docker pull webgoat/webgoat-8.0
docker run -p 8080:8080 -it webgoat/webgoat-8.0 /home/webgoat/start.sh
Firefox visit: http://127.0.0.1:8080/WebGoat
Create a user in WebGoat and log in.
In the WebGoat intro, I submit an email (no error, or any message to <user>@test.com
Then -
WebWolf (with WebGoat docker running)
docker pull webgoat/webwolf
docker run -p 8081:8081 -it webgoat/webwolf /home/webwolf/start.sh
Firefox visit: http://172.17.0.1:8081/login
   I try to log in with the cred used in WebGoat, to no avail, says unknown user. 
   I register a user with the same credentials as WebGoat, login, there is no email.

Should I be doing something different?
Thank you.

**Comment by nbaars** [comment]
@brohan: just to make sure, do you have WebGoat and WebWolf running at the same time? The e-mail cannot be delivered otherwise.

**Comment by brohan** [comment]
@nbaars Yes. I tried starting WebGoat, creating my user, sending the email, then freshly starting WebWolf and logging in using existing creds, which didn't work, then creating a user with the same creds.  Next I purged all docker data, re pulled, started both WebGoat and WebWolf, created new user in WebGoat, had to create user in WebWolf, then attempted to send the email.

**Comment by brohan** [comment]
Is there anything that you recommend trying to resolve this? Perhaps a specific sequence of events?

**Comment by nbaars** [comment]
@brohan can you explicitly pass the host and port to the WebGoat Docker image:

WebGoat determines the host and port of WebWolf as follows:

```
webwolf.host=${WEBWOLF_HOST:localhost}
webwolf.port=${WEBWOLF_PORT:8081}
```
So the default is localhost:8081. Think this might not be enough while running in Docker. Can you try to pass:

```
docker run -p 8080:8080 -it webgoat/webgoat-8.0 /home/webgoat/start.sh --env WEBWOLF_HOST=172.17.0.1
```

Let me know I will try to setup some tests myself with Docker.

**Comment by brohan** [comment]
My steps:

purged all Docker contents, as well as removed all cache / cookies from Firefox
docker pull webgoat/webgoat-8.0
docker run -p 8080:8080 -it webgoat/webgoat-8.0 /home/webgoat/start.sh --env WEBWOLF_HOST=172.17.0.1

New console:
docker pull webgoat/webwolf
docker run -p 8081:8081 -it webgoat/webwolf /home/webwolf/start.sh

In Firefox visited 127.0.0.1:8080/WebGoat
Registered a new user: user123 / user123
Jumped right to the WebWolf / Send email step, entered user123@test.com and submitted.

In Firefox visited 127.0.0.1:8081/WebWolf
Tried to login with creds user123 / user123, I received the error message "Invalid username or password"
I then register the same user via webwolf, was able to log in, however I did not see any email.

Should I complete the steps in a different order?

**Comment by nbaars** [comment]
@brohan yes, the old version still contained a check if the user exists within WebWolf. I've removed that checked based on this issue(not yet released). 

Can you try to first register the user within WebWolf?

Let me know if this works I can create a FAQ and update the lesson. 

Thanks for your help.

**Comment by brohan** [comment]
That order of operations appeared to not work as well.

Purged all docker items
Console #1:
docker pull webgoat/webwolf
docker run -p 8081:8081 -it webgoat/webwolf /home/webwolf/start.sh
Visit: http://172.17.0.1:8081/login
    Created a new user: user123 / user123! (this time was told my password did not contain the necessary characters, so I added the "!" to the end. This is different behavior than previous attempts where I created the user in WebGoat first then recreated in WebWolf with the creds: user123 / user123

Console #2:
docker pull webgoat/webgoat-8.0
docker run -p 8080:8080 -it webgoat/webgoat-8.0 /home/webgoat/start.sh --env WEBWOLF_HOST=172.17.0.1
Visit: http://127.0.0.1:8080/WebGoat
    Entered credentials created in WebWolf: user123 / user123!, which were not accepted. 
In WebGoat created user with the same creds: user123 / user123!, logged in, sent the email, which was not picked up in WebWolf.

I am up for testing any other combinations.

**Comment by nbaars** [comment]
Have something running with different command: (--env cannot be at the end)

```
docker run -p 8080:8080 --env WEBWOLF_HOST=172.17.0.1 -it webgoat/webgoat-8.0 /home/webgoat/start.sh
```

I can now send emails but fails due to the fact the user does not exists yet, which is correct.
Can you try this one?

Thanks

**Comment by brohan** [comment]
SUCCESS!  I was able to follow the same flow of commands as before, using your updated command to start WebWolf. I 1st created a user in WebWolf, then started WebGoat with the new command. My credentials for WebWolf did NOT work in WebGoat, however I registered the same creds in WebGoat, logged in, and sent the email. This time I received notice in WebGoat that the email was sent successfully. I was then able to see the email in WebWolf.

**Process:**
Purged all docker items
_Console #1:_
docker pull webgoat/webwolf
docker run -p 8081:8081 -it webgoat/webwolf /home/webwolf/start.sh
Visit: http://172.17.0.1:8081/login
Created a new user in WebWolf, then logged in.

Console #2:
docker pull webgoat/webgoat-8.0
docker run -p 8080:8080 --env WEBWOLF_HOST=172.17.0.1 -it webgoat/webgoat-8.0 /home/webgoat/start.sh
Visit: http://127.0.0.1:8080/WebGoat
Entered credentials created in WebWolf which were not accepted.
In WebGoat created user with the same creds as WebWolf. Logged in to WebGoat, sent an email (received notice that email was successfully sent).
Viewed the email in WebWolf.

Thank you for your patience in working through this.

**Comment by nbaars** [comment]
Great!! I will extend the lesson and create a FAQ how to get this working. I already created a docker-compose file to make this a little bit easier.

Thanks for the help,
--Nanne

**Comment by nbaars** [comment]
Remains open for extending the documentation

**Comment by nbaars** [comment]
Updated the lesson for starting WebWolf through Docker
